Provide control over padding in RtpPacketBuilder

dholroyd opened the pull request:
The options were previously:
 - pad to 4 bytes
 - no padding

This alters `RtpPacketBuilder::padding()` to allow other padding strategies to be supported:

 - `Pad::none()` - replaces `RtpPacketBuilder::padding(false)`
 - `Pad::round_to(n)` - replaces `RtpPacketBuilder::padding(true)` if `n == 4 `, and also allows for padding to other boundaries by specifying other values for `n`
 - `Pad::add_exactly(n)` manual control over the number of padding bytes to be added.  Maybe too much - I might remove this before release

I don't know of any scenario where padding to other than 4 bytes is needed, but the RTP syntax allows arbitrary padding and this change just seeks to expose that capability.

@WolverinDEV since you initially added the builder code: does your usage include padding, and would this alteration to the API cause you any problems?

dholroyd committed:
Provide greater control over output padding

WolverinDEV commented on src/builder.rs:
```diff
@@ -209,10 +287,10 @@ impl<'a> RtpPacketBuilder<'a> {
         }
 
-        if self.padded && (write_index & 0x3) != 0 {
+        if self.padded.is_padded() {
```
You can't simplify this check.
Imagine this scenario:  
- Set your padding to `Pad::round_to(4)`
- Set the payload of the packet with a length of a multiple of 4.
This will result in setting the padding flag even though we've appended zero bytes of padding.
This will cause an invalid packet.

I would suggest a `should_pad(write_index)` method here.

dholroyd replied:
Thanks!  I'll get this fixed.

WolverinDEV reviewed (changes requested):
I like the idea and how it's done.
Just fix the one issue and you're good to go :)

WolverinDEV commented:
> @WolverinDEV since you initially added the builder code: does your usage include padding, and would this alteration to the API cause you any problems?

Yes indeed this would cause a minor refactoring of my mcu code.  
You may want to increment the minor version by one.  
  
But in general I fully approve the change and think it adds something usefull.

dholroyd committed:
Only set padding flag if padding actually needed

dholroyd committed:
Make the assertion stronger

dholroyd committed:
Remove Pad::add_exactly()

If it turns out to be useful it can easily be added back in.  For now it seems
that the other available options should cover expected usage.
